jgrapht-core/src/test/java/org/jgrapht/alg/GabowStrongConnectivityInspectorTest.java

```diff
@@ -0,0 +1,285 @@
+/* ==========================================
+ * JGraphT : a free Java graph-theory library
+ * ==========================================
+ *
+ * Project Info:  http://jgrapht.sourceforge.net/
+ * Project Creator:  Barak Naveh (http://sourceforge.net/users/barak_naveh)
+ *
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+/* ------------------------------
+ * ConnectivityInspectorTest.java
+ * ------------------------------
+ * (C) Copyright 2003-2008, by Barak Naveh and Contributors.
+ *
+ * Original Author:  Barak Naveh
+ * Contributor(s):   John V. Sichi
+ *
+ * $Id$
+ *
+ * Changes
+ * -------
+ * 07-Aug-2003 : Initial revision (BN);
+ * 20-Apr-2005 : Added KosarajuStrongConnectivityInspector test (JVS);
+ *
+ */
+package org.jgrapht.alg;
+
+import junit.framework.TestCase;
+import org.jgrapht.DirectedGraph;
+import org.jgrapht.EdgeFactory;
+import org.jgrapht.VertexFactory;
+import org.jgrapht.alg.interfaces.StrongConnectivityInspector;
+import org.jgrapht.generate.RingGraphGenerator;
+import org.jgrapht.graph.*;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+/**
+ * Test cases for the GabowStrongConnectivityInspector. Tests are identical to the tests for the KosarajuStrongConnectivityInspector as provided in the
+ * ConnectivityInspectorTest class.
```

Comment:
Do you have a recommendation on how to do this? I agree that this would be cleaner, but looking at the test cases there are quite a lot of different instances being created of the various StrongConnectivityInspectors. I can't just pass a connectivityInspector as a parameter because you need to specify the graph in the constructor. So I don't really know how to do this.